src/Tasks/Services/ProcessProducts.php

```diff
@@ -278,78 +266,94 @@ private function processWineVariation(ScheduledWineVariation $scheduledWineVaria
+     * @param                        $productAttribute
+     * @return \SilverStripe\ORM\DataObject|ProductAttributeOption
+     */
+    private function createProductAttributeOption(ScheduledWineVariation $scheduledWineVariation, $productAttribute)
+    {
+        // get the product attribute option if it exists
+        $productAttributeOption = ProductAttributeOption::get()
+            ->filter([
+                'Title'              => $scheduledWineVariation->Title,
+                'ProductAttributeID' => $productAttribute->ID
+            ])->first();
 
         // Format the price modifier amount
         $modifier = $scheduledWineVariation->PriceModifierAmount;
-
         if ($modifier > 0) {
             $modifier = str_replace('.', '', $modifier);
         }
 
+        // check if it already exists then update the price
         if ($productAttributeOption && $productAttributeOption->exists()) {
-            // Ensure price is updated
             $productAttributeOption->setField('SKU', $scheduledWineVariation->SKU);
             $productAttributeOption->setField('PriceModifierAmount', $modifier);
-
             $productAttributeOption->write();
-
-            $productAttributeOptionID = $productAttributeOption->ID;
         } else {
-            // Create it
+            // If it doesnt exist then create it
             $productAttributeOption = ProductAttributeOption::create();
-
             $productAttributeOption->update([
                 'ClassName'             => ProductAttributeOption::class,
                 'Title'                 => $scheduledWineVariation->Title,
-                'ProductAttributeID'    => $vintageAttribute->ID,
+                'ProductAttributeID'    => $productAttribute->ID,
                 'PriceModifierAmount'   => $modifier,
                 'PriceModifierCurrency' => 'NZD' // TODO: Use dynamic currency
             ]);
-
-            $productAttributeOptionID = $productAttributeOption->write();
+            $productAttributeOption->write();
         }
 
+        // publish updates
         $productAttributeOption->publishSingle();
 
+        // remove attribute where SKU matches and product attribute is different
+
+        // return product attribute option
+        return $productAttributeOption;
+    }
+
+    /**
+     * @param ScheduledWineVariation $scheduledWineVariation
+     * @param WineProduct            $wineProduct
+     * @param ProductAttributeOption $productAttributeOption
+     */
+    private function createProductVariationOptions(ScheduledWineVariation $scheduledWineVariation, WineProduct $wineProduct, ProductAttributeOption $productAttributeOption)
+    {
+
         // Get the options
         /** @var ComplexProductVariation_Options $variationOptions */
         $variationOptions = ComplexProductVariation_Options::get()
-            ->filter('ProductAttributeOptionID', $productAttributeOptionID)
+            ->filter('ProductAttributeOptionID', $productAttributeOption->ID)
```

Comment:
This won't work if you have multiple attributes (e.g. pack, wines).

This is why this code worked in the past; With ONLY vintage, it was fine.

If multiple pack sizes have the same vintage, this code will pick one of them randomly!

The good news is that WineProduct::updatePacksForVintage() will fill out the rest of the product variations for each vintage.

The bad news is that flow (i.e. this code) will need to setup that $firstVariation for each vintage.

The solution in this case is to create not ONLY the wine attribute, but also the pack size attribute at the same time (any valid pack size will do).

Then this code should find the ComplexProductVariation based on BOTH the pack size and the vintage.

You don't really need the ComplexProductVariation_Options itself, in this case. you just need the variation right?

You'll also need to pass a new parameter in for this method:
 - rename $productAttributeOption to $productVintageOption
- add $productPackOption